pkg/sql/sem/builtins/generator_builtins.go

```diff
@@ -228,6 +264,38 @@ func makeSeriesGenerator(_ *tree.EvalContext, args tree.Datums) (tree.ValueGener
 	}, nil
 }
 
+// ResolvedType implements the tree.ValueGenerator interface.
+func (*seriesTSValueGenerator) ResolvedType() types.TTable { return seriesTSValueGeneratorType }
+
+// Start implements the tree.ValueGenerator interface.
+func (s *seriesTSValueGenerator) Start() error { return nil }
+
+// Close implements the tree.ValueGenerator interface.
+func (s *seriesTSValueGenerator) Close() {}
+
+// Next implements the tree.ValueGenerator interface.
+func (s *seriesTSValueGenerator) Next() (bool, error) {
```

Comment:
This code seems exactly the same as the other series generator code. Perhaps there's a way to modify the seriesValueGenerator to parameterize it based on type. It looks like ResolvedType and Values would need to check the type parameter, but everything else could be the same.

Reply:
ok, i'll try